tests/test_local_deployment.py

```diff
@@ -215,6 +232,35 @@ def test_load_balancing(local_deployment, query):
     assert result
 
 
+@pytest.mark.local
+@pytest.mark.parametrize("enable_restful_api", [True])
+@pytest.mark.parametrize("restful_api_port", [28080])
+@pytest.mark.parametrize(
+    "task_name, model_name, query",
+    [
+        (
+            "text-generation",
+            "bigscience/bloom-560m",
+            {
+                "query": ["DeepSpeed is the greatest"]
+            },
+        ),
+    ],
+)
+def test_restful_api(local_deployment, query, restful_api_port):
+    generator = mii.mii_query_handle(local_deployment.deployment_name)
+    for _ in range(2):
+        result = generator.query(query)
+
+    url = f'http://localhost:{restful_api_port}/mii/{local_deployment.deployment_name}'
+    params = {"request": query}
+    json_params = json.dumps(params)
+    result = requests.post(url,
+                           data=json_params,
+                           headers={"Content-Type": "application/json"})
+    assert result
```

Comment:
We probably want to assert that the result contains a response - I think this would pass on an error if the `requests.post` returned an error message.